### PR TITLE
Environment variables are moved to shared config utility

### PR DESCRIPTION
**Pull request by sunhay**
Environment variables now override values in the shared config package.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -164,10 +162,6 @@ func NewDefaultAgentConfig() *AgentConfig {
 		StatsdHost: "127.0.0.1",
 		StatsdPort: 8125,
 
-		// Path and environment for the dd-agent embedded python
-		DDAgentPy:    defaultDDAgentPy,
-		DDAgentPyEnv: []string{defaultDDAgentPyEnv},
```
Comment (sunhay) [inline config/config.go]: Removing these as we don't need them anymore

File: config/config.go [hunk 2]
```diff
@@ -222,20 +216,41 @@ func NewAgentConfig(yamlPath, netYamlPath string) (*AgentConfig, error) {
 
 	// For Agent 6 we will have a YAML config file to use.
 	if util.PathExists(yamlPath) {
```
Comment (sfluor) [inline config/config.go]: Should we log an error if this is not the case now that we support only yaml config ?

Comment (sunhay) [reply config/config.go]: Technically you can start the agent without a config. I'll log a warning though 👍

Comment (sfluor) [reply config/config.go]: oh but then it could be ran by the network-tracer and not exist nevermind

File: config/config.go [hunk 3]
```diff
@@ -289,12 +309,19 @@ func NewNetworkAgentConfig(yamlPath string) (*AgentConfig, error) {
 	}
 
 	if util.PathExists(yamlPath) {
-		if err := cfg.loadNetworkYamlConfig(yamlPath); err != nil {
+		config.Datadog.AddConfigPath(yamlPath)
```
Comment (sfluor) [inline config/config.go]: should we turn this segment and the one above into a function since it's used twice ?

File: config/config.go [hunk 4]
```diff
@@ -304,138 +331,52 @@ func NewNetworkAgentConfig(yamlPath string) (*AgentConfig, error) {
...
+		"DD_PROXY_HTTPS":    "proxy.https",
+
+		"DD_LOGS_STDOUT":    "log_to_console",
+		"LOG_TO_CONSOLE":    "log_to_console",
+		"DD_LOG_TO_CONSOLE": "log_to_console",
+		"LOG_LEVEL":         "log_level", // Support LOG_LEVEL and DD_LOG_LEVEL but prefer DD_LOG_LEVEL
+		"DD_LOG_LEVEL":      "log_level",
+	} {
+		if v, ok := os.LookupEnv(envKey); ok {
+			config.Datadog.Set(cfgKey, v)
```
Comment (sfluor) [inline config/config.go]: I thought the goal was to make `Set` private did it changed ?

Comment (sunhay) [reply config/config.go]: Yeah, I thought so too. I'm not sure how we'd properly handle environment variables then.

I'm just going to follow the trace-agent's example for now and we can change in the future depending on what happens downstream.

**Pull request merged**